docker/cli
# Add 'docker service rollback' subcommand

Signed-off-by: Jimmy Leger <jimmy.leger@gmail.com>

fixes #142
PS: This bug was fixed during a Docker Hackergarten. 

**- What I did**

I add the docker service rollback subcommand

**- How I did it**

I extracted the rollback process from **runUpdate**

**- How to verify it**

Create a service with a single replica:

```bash
$ docker service create --name my-service -p 8080:80 nginx:alpine
```

Confirm that the service is running with a single replica:

```bash
$ docker service ls

ID                  NAME                MODE                REPLICAS            IMAGE               PORTS
xbw728mf6q0d        my-service          replicated          1/1                 nginx:alpine        *:8080->80/tcp
```

Update the service to use three replicas:

```bash
$ docker service update --replicas=3 my-service

$ docker service ls

ID                  NAME                MODE                REPLICAS            IMAGE               PORTS
xbw728mf6q0d        my-service          replicated          3/3                 nginx:alpine        *:8080->80/tcp
```

Now roll back the service to its previous version, and confirm it is
running a single replica again:

```bash
$ docker service rollback my-service

$ docker service ls

ID                  NAME                MODE                REPLICAS            IMAGE               PORTS
xbw728mf6q0d        my-service          replicated          1/1                 nginx:alpine        *:8080->80/tcp
```

**- Description for the changelog**

Add docker service rollback subcommand

**- A picture of a cute animal (not mandatory but encouraged)**

![redpanda picture](http://elelur.com/data_images/mammals/red-panda/red-panda-05.jpg)
